### PR TITLE
Add url_fetch, lock, properties, cache. Updated tasks with an options factory

### PR DESCRIPTION
**Pull request by CorvetteCole**
I am in the middle of a cool little project utilizing these here: https://github.com/CorvetteCole/CanvasTasks.

Still in the messy stage but I've been very impressed with how far Dart can go on Google Apps Scripts.

Only thing is need to use the new runtime because otherwise you get UInt8Array errors.

### CODE REVIEW

File: lib/tasks.dart
Comment (floitsch) [inline lib/tasks.dart]:
```suggestion
  // Figure out how to seamlessly convert these to DateTimes for dart.
```

If the API returns a String, it's probably best to just go with it, and transform it in user-code (potentially providing a helper function).
At some point I tried to dartify the APIs, and it was just too much work. It's easier to live with the original API...

If the API returns a `Date` object, it should automatically be transformed to a `DateTime` (I believe).

File: lib/url_fetch.dart
Comment (floitsch) [inline lib/url_fetch.dart]:
```suggestion
```

File: lib/url_fetch.dart
Comment (floitsch) [inline lib/url_fetch.dart]: Add new line.

File: lib/properties.dart
Comment (floitsch) [inline lib/properties.dart]:
```suggestion
```

**Review by floitsch — APPROVED** [review]
LGTM!
Thanks.

**Comment by floitsch** [comment]
For information, the typical work-flow is:
If approved, go through the comments and address them.
If you agree, fix them, and then merge it. (If you have the permissions. If not, please let me know).

If you disagree, comment on it, and don't merge yet. (Gives me the opportunity to respond).
Hope that makes sense.

**Comment by floitsch** [comment]
I haven't yet played with the new runtime. (Should be dramatically faster).
Is this something that happens automatically, or do the Dart tools need to be adjusted?

**Comment by CorvetteCole** [comment]
Thanks for the suggestions and info. I'm relatively new to making contributions so I appreciate it! The Dart tools do not have to be adjusted. The only issue I've had is that the debugging does not seem to work properly on large files in the new runtime, so you have to switch to the old runtime and then switch back when you've solved your problem.

I'll go through these conversations and get things worked out. Thanks again

**Comment by floitsch** [comment]
I actually had to abandon the debugger for my bigger scripts: https://github.com/google/dart_google_apps/tree/master/example/tournament

Ended up doing print-debugging...

**Comment by CorvetteCole** [comment]
Same here with the print debugging... Although it was nice to be pointed in the right direction sometimes by the debugging console. Also found that the apps_scripts_tools main watch method does not successfully upload new script versions, at least not in a reasonable amount of time when they get big. I've taken to copy and pasting the .gs file contents. 

It would be really really cool to allow people to write Dart in Google Apps Scripts natively. It feels like that is a perfect use case for Dart.

**Comment by CorvetteCole** [comment]
I went ahead and made those minor changes. I will need you to merge this though since it requires write access. Thanks for maintaining this repo, I love Dart :)

**Comment by floitsch** [comment]
Thanks again.

**Pull request merged**
